internal/appconfig/validation.go

```diff
@@ -275,6 +275,14 @@ func (cfg *Config) validateMounts() (extraInfo string, err error) {
 			}
 		}
 
+		switch {
+		case m.SnapshotRetention == 0:
+			m.SnapshotRetention = 5
```

Comment:
hmm, mutating the configuration while validating it is not the right place IMHO.

Isn't `0` a valid retention value in case the use wants to disable snapshots?

If the problem is that CreateVolumeRequest uses an `*int` type for SnapshotRetention, I would vote to use the same type for `Mount.SnapshotRetention`. It makes things much easier to propagate and let the API service decide what value to use by default.

Reply:
thx, i like that better!